### PR TITLE
New Alviy provider

### PR DESCRIPTION
**Pull request by alviy**
Hello Acme Team! 

Kindly ask to add Alviy DNS provider. All DNS tests are passed. It was hard work :-/

### CODE REVIEW

**Comment by alviy** [comment]
Test https://github.com/alviy/acme.sh/actions/runs/8859286855
Docs https://github.com/acmesh-official/acme.sh/wiki/dnsapi2#dns_alviy

**Comment by alviy** [comment]
@Neilpang could you review this PR please?

Thanks a lot!

File: dnsapi/dns_alviy.sh [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+}
+
+####################  Private functions below ##################################
+#_acme-challenge.www.domain.com
+#returns
+# _sub_domain=_acme-challenge.www
+# _domain=domain.com
+_get_root() {
+  domain=$1
+  i=3
```
Comment (Neilpang) [inline dnsapi/dns_alviy.sh:104]: why does i start from 3 ?
please start from 1

**Comment by alviy** [comment]
@Neilpang could you please look at the commit comments?

File: dnsapi/dns_alviy.sh (hunk 1 above)
Comment (Neilpang) [reply dnsapi/dns_alviy.sh:104]: can you process all the formats here:
https://github.com/acmesh-official/acme.sh/wiki/DNS-API-Dev-Guide#3-detect-which-part-is-your-root-zone


image

**Pull request merged**